datastore/longhorn.go

```diff
@@ -242,14 +242,14 @@ func checkVolume(v *longhorn.Volume) error {
 	if v.Name == "" || size == 0 || v.Spec.NumberOfReplicas == 0 {
 		return fmt.Errorf("BUG: missing required field %+v", v)
 	}
+	if len(v.Name) > NameMaximumLength {
+		logrus.Warnf("Volume name is too long %v, auto-correct to %v characters", v.Name, NameMaximumLength)
+		v.Name = strings.TrimRight(v.Name[:NameMaximumLength], "-")
+	}
```

Comment:
same issue here, feels like this could end up with name collisions?

Reply:
See https://github.com/longhorn/longhorn-manager/pull/825#discussion_r570705312.